Modules/main.c

```diff
@@ -549,7 +597,7 @@ pymain_repl(PyConfig *config, int *exitcode)
         *exitcode = (run != 0);
         return;
     }
-    int run = pymain_run_module(L"_pyrepl", 0);
+    int run = pymain_start_pyrepl_no_main();
```

Comment:
My first intuition was to simply `PyImport_ImportModule("_pyrepl.__main__")` but then the entire shell session is a side-effect of an on-going import, which was ugly and caused some funny edge cases (a non-zero exit was technically an ImportError).

So instead we do the full dance of running `_pyrepl.main.interactive_console()`. While doing that I'm smuggling PYTHONSTARTUP support because:
- you can always remove PYTHONSTARTUP from your environment to not have it;
- but previously you could not get PYTHONSTARTUP to be executed even if you wanted.